modules/dnn/src/layers/einsum_layer.cpp

```diff
@@ -105,15 +105,15 @@ static Mat batchwiseMatMul(
 
         // input1 should of size MxK
         // check if input1 needs reshape, if need reshape
-        if (input1.dims > 2 || input1.size[0] != M || input1.size[1] != K)
+        if (input1.dims > 2 || input1.size[0] != M || (input1.dims > 1 && input1.size[1] != K) || input1.dims == 1)
```

Comment:
To make 4.x/5.x more compatible, can you propose `einsum_layer.cpp` change to 4.x branch too?

Reply:
Without enabling tests